sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -258,12 +300,14 @@ def test_spectral_embedding_amg_solver(seed=36):
 @pytest.mark.filterwarnings(
     "ignore:scipy.linalg.pinv2 is deprecated:DeprecationWarning:pyamg.*"
 )
-def test_spectral_embedding_amg_solver_failure():
+@pytest.mark.parametrize("dtype", ("'float32'", "'float64'"))
```

Comment:
This should solve the `astype` error

```suggestion
@pytest.mark.parametrize("dtype", (np.float32, np.float64))
```

Reply:
I have tried it first - gives an error. Now testing
@pytest.mark.parametrize("dtype", ("float32'", "'float64"))